include/core/physics_solver.h

```diff
@@ -114,12 +114,36 @@ class PhysicsSolver
    * @brief Default way to evaluate the residual for the nonlinear solver.
    * Some application may use more complex evaluation of the residual and
    * override this method.
+   *
+   * @params[in] normalize_residual_by_volume If true, the residual is
+   * normalized by the global volume of the triangulation.
    */
   virtual double
-  get_current_residual()
+  get_current_residual(const bool &normalize_residual_by_volume = false)
   {
+    const double normalize_volume =
+      normalize_residual_by_volume ? get_global_volume() : 1.0;
     auto &system_rhs = get_system_rhs();
-    return system_rhs.l2_norm();
+    return system_rhs.l2_norm() / normalize_volume;
+  }
+
+
+  /**
+   * @brief Return varialbe with volume of the entire triangulation.
+   */
+  inline double
+  get_global_volume() const
+  {
+    return global_volume;
+  }
+
+  /**
+   * @brief Set variable holding the volume of the entire triangulation.
+   */
+  inline void
+  set_global_volume(const double &volume)
+  {
+    global_volume = volume;
   }
```

Comment:
Why instead of a setter and a getter you don't force a virtual void function which would be something like get_domain_metric or something like that?